samd/pins.h

```diff
@@ -30,7 +30,50 @@
 #ifndef MICROPY_INCLUDED_ATMEL_SAMD_PERIPHERALS_PINS_H
 #define MICROPY_INCLUDED_ATMEL_SAMD_PERIPHERALS_PINS_H
 
-#include "mpconfigport.h"
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "include/sam.h"
+
+#include "samd_peripherals_config.h"
+
+typedef struct {
+    uint8_t index:6;            // 0, 1, etc. corresponding to SERCOM<n>.
+    uint8_t pad:2;              // which of the four SERCOM pads to use
+} pin_sercom_t;
+
+typedef struct {
+    uint8_t index:4;
+    bool is_tc:1;
+    uint8_t wave_output:3;
+} pin_timer_t;
+
+#ifdef SAMD21
+    #define NUM_TIMERS_PER_PIN 2
+    #define NUM_ADC_PER_PIN 1
+#endif
+#ifdef SAMD51
+    #define NUM_TIMERS_PER_PIN 3
+    #define NUM_ADC_PER_PIN 2
+#endif
+#define NUM_SERCOMS_PER_PIN 2
+
+#ifndef PIN_PREFIX_FIELDS
+#define PIN_PREFIX_FIELDS
+#define PIN_PREFIX_VALUES
+#endif
```

Comment:
I'm assuming you are defining these in the `circuitpython` repo. Could you add some comments here to show what these `#define`s are for, so someone using this library independently will have some guidance?